### PR TITLE
WT-2290: don't reuse previous multi-block write blocks when compacting

### PR DESCRIPTION
**Pull request by keithbostic**
@agorrod, @sue, ready for review/merge.

### CODE REVIEW

File: test/suite/test_compact02.py [hunk 1]
```diff
@@ -115,11 +115,9 @@ def test_compact02(self):
...
         self.pr('Removed total ' + str((count * 9666) / mb) + 'MB')
 
-        # 4. Reopen the connection to force the file to disk (if we leave dirty
-        # blocks in the cache, it can affect how compact works depending on how
-        # and when those blocks are written, because they're being written best
-        # fit, not first-fit, as compaction does).
-        self.reopen_conn()
+        # 4. Checkpoint to make the deleted records available for reuse, so
+        # compaction can be successful.
+        self.session.checkpoint()
```
Comment (sueloverso) [inline test/suite/test_compact02.py]: I don't think this is necessary.  The test runs without it (I ran 50 iterations on the AWS box that was failing reliably).  The code in `compact_file` calls checkpoint before starting and on each pass already.  I also think not having it is more realistic for what a user is going to do.  Did you see a failure without it?

File: src/include/session.h [hunk 2]
```diff
@@ -74,7 +74,10 @@ struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_session_impl {
 	TAILQ_HEAD(__cursors, __wt_cursor) cursors;
 
 	WT_CURSOR_BACKUP *bkp_cursor;	/* Hot backup cursor */
-	WT_COMPACT	 *compact;	/* Compact state */
+
+	WT_COMPACT	 *compact;	/* Compaction information */
+	enum { WT_COMPACT_NONE=0,
+	    WT_COMPACT_RUNNING, WT_COMPACT_SUCCESS } compact_state;
```
Comment (sueloverso) [inline src/include/session.h]: Do we still need `session->compaction`?  If not, I don't see its removal here.

**Comment by sueloverso** [comment]
@keithbostic This LGTM.  I had a couple minor comments/questions.

File: test/suite/test_compact02.py (hunk 1 above)
Comment (keithbostic) [inline test/suite/test_compact02.py]: I agree, this probably isn't necessary, I'll pull it out, as well as the subsequent stat call.

File: src/include/session.h (hunk 2 above)
Comment (keithbostic) [inline src/include/session.h]: Thanks! Change pushed.

**Pull request merged**